packages/health-check/src/verboseChecks.js

```diff
@@ -551,6 +370,59 @@ function uniswapIndexFileCheck(done) {
   skylinkVerification(done, linkInfo);
 }
 
+// uniswapHNSCheck returns the result of trying to download the skylink
+// for the Uniswap Application with the HNS domain
+function uniswapHNSCheck(done) {
+  const linkInfo = {
+    description: "Uniswap HNS",
+    skylink: "hns/doesn/",
+    bodyHash: uniswapBodyHash,
+    metadata: uniswapMetadata,
+  };
+
+  skylinkVerification(done, linkInfo);
+}
+
+// uniswapHNSRedirectCheck returns the result of trying to download the skylink
+// for the Uniswap Application with the HNS domain and without a trailing slash
+function uniswapHNSRedirectCheck(done) {
+  const linkInfo = {
+    description: "Uniswap HNS Redirect",
+    skylink: "hns/doesn",
+    bodyHash: uniswapBodyHash,
+    metadata: uniswapMetadata,
+  };
+
+  skylinkVerification(done, linkInfo);
+}
+
+// uniswapHNSResolverCheck returns the result of trying to download the skylink
+// for the Uniswap Application via the HNS resolver endpoint
+function uniswapHNSResolverCheck(done) {
+  const linkInfo = {
+    description: "Uniswap HNS Resolver",
+    skylink: "hnsres/doesn/",
+    bodyHash: "44a3f0f56861ae841a6cb19cb0b3edf98ad610f8",
+    metadata: null,
+  };
+
+  skylinkVerification(done, linkInfo);
+}
+
+// uniswapHNSResolverRedirectCheck returns the result of trying to download the skylink
+// for the Uniswap Application via the HNS resolver endpoint without the
+// trailing slash
+function uniswapHNSResolverRedirectCheck(done) {
+  const linkInfo = {
+    description: "Uniswap HNS Resolver Redirect",
+    skylink: "hnsres/doesn",
+    bodyHash: "44a3f0f56861ae841a6cb19cb0b3edf98ad610f8",
+    metadata: null,
```

Comment:
@kwypchlo these don't have skyfile metadata defined so the response is `undefined`. Will setting this to `null` work with the `lodash.IsEqual`?

Reply:
Nope, but you might want to set `metadata: undefined` and it should work in theory :)

Reply:
Why doesn't it have the metadata though? Is this correct?